src/components/Messages.tsx

```diff
@@ -3,21 +3,31 @@
 import {FC, useRef, useState} from "react";
 import {Message} from "@/lib/validations/messages"
 import {MessageTimestamp} from "@/components/MessageTimestamp";
+import MessageThumbnail from "@/components/MessageThumbnail";
 
 interface MessagesProps {
     initialMessages: Message[],
+    participants: ChatParticipants,
+}
+
+interface ChatParticipants{
+    user: User
+    partner: User
     sessionId: string
 }
 
-const Messages: FC<MessagesProps> = ({initialMessages, sessionId}) => {
+const Messages: FC<MessagesProps> = ({initialMessages, participants}) => {
     const [messages] = useState<Message[]>(initialMessages)
     const scrollDownRef = useRef<HTMLDivElement | null>(null)
 
     return <div aria-label='messages' className='message-scroll'>
         <div ref={scrollDownRef}>
         {messages.map((message, index) => {
             const hasNextMessage = userHasNextMessage(messages, index)
-            const classes = new ClassNames(sessionId === message.senderId, hasNextMessage)
+            const currentUser = participants.sessionId === message.senderId
+            const classes = new ClassNames(currentUser, hasNextMessage)
+            const userName = currentUser ? participants.user.name : participants.partner.name
```

Comment:
timmy two-touch and too many ternaries:

update this to an if-statment - it will also break up the text into a human-readable flow